application/helpers/admin/import_helper.php

```diff
@@ -2416,7 +2416,7 @@ function TSVImportSurvey($sFullFilePath)
                     }
                     // Insert sub question and keep the sqid for multi language survey
                     $question = new Question();
-                    $question->attributes = $insertdata;
+                    $question->setAttributes($insertdata,false); //$question->attributes = $insertdata; : this broke call of $this when test rules when save : PLEASE : don't use
```

Comment:
By-passing validation is not a real fix. I still prefer Dominik approach: at least by reading his code any we understand what the problem is and what the fix is doing. 
I'd say a "real" fix would be to change the subquestion's code generation for ranking question  so it respects the rule (what dominik started to do), or to change the rule.

Reply:
I don't bypass validation … I respect the rules : see the bug report PLEASE ! 

And yes : i'm angry here …

Reply:
Only two attributes are set to safe, no? https://github.com/LimeSurvey/LimeSurvey/blob/master/application/models/Question.php#L114

Reply:
Sorry, I'm wrong. All attributes that have a validation rule are safe in Yii.

Reply:
Ok, we reviewed everything a last time here. The problem is that exporting to tsv create subquestion being wrong in a way or another making the question rules to apply. Dominik will do a "real" fix today: he will call the lss generator (which works fine), and convert the result to TSV. So we'll have a single logic for both format.

Reply:
And there are the second issue : even valid TSV didn't import …

Reply:
There are multiple other issues with TSV export/import, for example number of question attributes being different (or maybe your fix fixed that too?).